File: algorithm/baekjoon/14JUN21/7_1157.py
#알파벳 대소문자로 된 단어가 주어지면, 이 단어에서 가장 많이 사용된 알파벳이 무엇인지 알아내는 프로그램을 작성하시오. 단, 대문자와 소문자를 구분하지 않는다.
#첫째 줄에 이 단어에서 가장 많이 사용된 알파벳을 대문자로 출력한다. 단, 가장 많이 사용된 알파벳이 여러 개 존재하는 경우에는 ?를 출력한다.

#input값을 다 대문자로 받음 > 출력을 대문자로 해야해서
#'hello'
input = input().upper()
#input값을 set으로 리스트를 만들어 중복된 알파벳을 삭제
#'h,e,l,o'
words = list(set(input))

#최빈값을 구하기 위해
cnt_list = []

for i in words:
    #input값에서 각 알파벳 카운팅
    #2(l),1,1,1
    cnt = input.count(i)
    #리스트에 붙여줌
    cnt_list.append(cnt)
    
#만약 cnt_list의 최대값이 1보다 많으면 = 중복되면
if cnt_list.count(max(cnt_list))>1:
    print('?')
#최빈값 = 최대값 인덱스
else :
    max_index = cnt_list.index(max(cnt_list))
    print(words[max_index])

File: algorithm/baekjoon/15JUN21/7_1157.js
//TODO
// let fs = require("fs");
// let input = fs.readFileSync("/dev/stdin").toString();

function Word(word) {
  const alphabet = new Array(26).fill(0);

  let charAry = word.toLowerCase().split("");
  let asciiA = "a".charCodeAt();

  //forEach : 배열의 요소를 반복
  charAry.forEach((char) => {
    let asciiChar = char.charCodeAt();
    let index = asciiChar - asciiA;
    alphabet[index] += 1;
  });

  //배열 [], 혹은 객체 {} 안의 값을 편하게 꺼내 쓸 수 있는 문법
  let maxNum = Math.max(...alphabet);
  let maxIndex = alphabet.indexOf(maxNum);
  let isSame = false;

  for (let i = 0; i < alphabet.length; i++) {
    if (maxNum === alphabet[i] && i !== maxIndex) {
      isSame = true;
    }
  }
  return isSame ? "?" : String.fromCharCode(maxIndex + asciiA).toUpperCase();
}

console.log(Word('hello'));
